### PR TITLE
Use ReactNative.V8JSI.Windows.0.1.6 and ReactNative.Hermes.Windows.0.1.6

### PR DESCRIPTION
**Pull request by licanhua**
Part of implementation #3620
Multiple version of V8JSI and Hermes are referenced in the project. and multiple version of them are downloaded even if we don't use them for the first time.
Unified them to one version 0.1.6, and it will save 1.3G disk

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/3729)

### CODE REVIEW

**Comment by acoates-ms** [comment]
@mganandraj can you look at this.  I assume we want them aligned, just want to check this is version we want.

File: vnext/ReactUWP/packages.config [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <packages>
   <package id="boost" version="1.68.0.0" targetFramework="native" />
   <package id="Microsoft.UI.Xaml" version="2.3.191129002" targetFramework="native" />
+  <package id="ReactNative.Hermes.Windows" version="0.1.6" targetFramework="native" Condition="'$(OSS_RN)' != 'true' AND '$(USE_HERMES)' == 'true'" />
```
Comment (mganandraj) [inline vnext/ReactUWP/packages.config]: If I remember correctly.. the predicates to check for e.g. USE_HERMES is true doesn't work .. as we use the oldish "package.config" way .. The predicates are supported with the PackageReference way (https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files ) but I had some other trouble which I don't remember exactly ...

File: vnext/ReactUWP/ReactUWP.vcxproj [hunk 2]
```diff
@@ -418,8 +418,8 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
     <Import Project="$(SolutionDir)packages\boost.1.68.0.0\build\boost.targets" Condition="Exists('$(SolutionDir)packages\boost.1.68.0.0\build\boost.targets')" />
-    <Import Project="$(SolutionDir)packages\ReactNative.Hermes.Windows.0.1.3\build\native\ReactNative.Hermes.Windows.targets" Condition="Exists('$(SolutionDir)packages\ReactNative.Hermes.Windows.0.1.3\build\native\ReactNative.Hermes.Windows.targets') AND '$(OSS_RN)' != 'true' AND '$(USE_HERMES)' == 'true'" />
-    <Import Project="$(SolutionDir)packages\ReactNative.V8JSI.Windows.0.1.4\build\native\ReactNative.V8JSI.Windows.targets" Condition="Exists('$(SolutionDir)packages\ReactNative.V8JSI.Windows.0.1.4\build\native\ReactNative.V8JSI.Windows.targets') AND '$(OSS_RN)' != 'true' AND '$(USE_V8)' == 'true'" />
+    <Import Project="$(SolutionDir)packages\ReactNative.Hermes.Windows.0.1.6\build\native\ReactNative.Hermes.Windows.targets" Condition="Exists('$(SolutionDir)packages\ReactNative.Hermes.Windows.0.1.6\build\native\ReactNative.Hermes.Windows.targets') AND '$(OSS_RN)' != 'true' AND '$(USE_HERMES)' == 'true'" />
```
Comment (mganandraj) [inline vnext/ReactUWP/ReactUWP.vcxproj]: This is the latest and best version .. and I was thinking we are already on 0.1.6 .. Thanks !

File: vnext/ReactUWP/packages.config (hunk 1 above)
Comment (mganandraj) [reply vnext/ReactUWP/packages.config]: Essentially, the msbuild variables are not visible in package.config ..

Comment (mganandraj) [reply vnext/ReactUWP/packages.config]: Fixing this will help in reducing the build times ..

File: vnext/ReactUWP/ReactUWP.vcxproj (hunk 2 above)
Comment (mganandraj) [reply vnext/ReactUWP/ReactUWP.vcxproj]: Actually, you should reference V8/Hermes package like we do in https://github.com/microsoft/react-native-windows/blob/master/vnext/ReactWindowsCore/ReactWindowsCore.vcxproj ..

My understanding was that, as ReactUWP links to ReactWindowsCore, these lines are not needed in ReactUWP ..

Comment (mganandraj) [reply vnext/ReactUWP/ReactUWP.vcxproj]: The common variable are defined in https://github.com/microsoft/react-native-windows/blob/master/vnext/PropertySheets/React.Cpp.props

Comment (licanhua) [reply vnext/ReactUWP/ReactUWP.vcxproj]: ok. I removed them from ReactUWP project.

File: vnext/ReactUWP/packages.config (hunk 1 above)
Comment (licanhua) [reply vnext/ReactUWP/packages.config]: I postpone this to #3753. it's possible to cover WinUI and cppwinrt nuget too

**Pull request merged**